### DISCUSSION TITLE
How are you using Laravel Love?

### DISCUSSION THREAD

**Discussion by antonkomarev** [cybercog/laravel-love#74] [Show and tell]
Please leave a comment explaining how you are using this package.
Any feedback will be helpful when developing the new version!

All these questions are optional, but might help to find bottlenecks of the package:
- How many users are you supporting?
- What type of features have you had to implement to extend the base functionality of this package?

*If you have any issues, please open up a new ticket. Please keep this thread for feedback only. Thanks!*

**Comment by iamB0rgy** [comment]
I'm using Laravel Love to heart and unheart my Recipe Post with the help of Ajax so it would appear instantly when users hearted the recipe post. 

On my Blade. 

                             @if ($hasReacted)
                                <div id="hearted">
                                    <a class="mb-0 px-0 py-1" href="#" id="liked">
                                        <span class="ms-1">{{ $reactionCounter }}</span>
                                        <input type="hidden" id="recipeId" value="{{ $recipe->id }}">
                                        <img class="x1b0d499 xep6ejk" src="{{ asset('img/icons/icons8-blue-heart-25.png') }}">
                                    </a>
                                </div>
                                 <div id="newHeartStatus"></div>
                                @else
                                <div id="unheart">
                                    <a class="mb-0 px-0 py-1" href="#" id="liked">
                                        <span class="ms-1">{{ $reactionCounter }}</span>
                                        <input type="hidden" id="recipeId" value="{{ $recipe->id }}">
                                        <img class="x1b0d499 xep6ejk" src="{{ asset('img/icons/icons8-white-heart-25.png') }}">
                                    </a>
                                </div>
                                <div id="newHeartStatus"></div>
                                @endif

Ajax;

      $(document).on('click','#liked',function(event){
        event.preventDefault();
        let recipeId = $("#recipeId").val();
        $.ajax({
            method: "POST",
            headers: {
                'X-CSRF-TOKEN': '{{ csrf_token() }}'
                },
            url : "{{ route('recipe.likes' , $recipe->id) }}",
            data:{
                'id': recipeId,
            },
            dataType: 'json',

        success : function(response) {
            if (response.success === true) {
                Swal.fire({
                    toast: true,
                    icon: response.icon,
                    title: response.title,
                    position: 'top',
                    showConfirmButton: false,
                    timer: 2000,
                    timerProgressBar: true,
                });
                if(response.status === 'Liked'){
                    $("#unheart").remove();
                    var html = '<div id="hearted">';
                        html += '<a class="mb-0 px-0 py-1" href="#" id="liked">';
                        html += '<span class="ms-1">'+ response.count +'</span>';
                        html += '<input type="hidden" id="recipeId" value="{{ $recipe->id }}">';
                        html += '<img class="x1b0d499 xep6ejk" src="{{ asset("img/icons/icons8-blue-heart-25.png") }}">';
                        html += '</a>';
                        html += '</div>';
                        html += '<div id="newHeartStatus"></div>';
                    $('#newHeartStatus').append(html);
                }else{
                    $("#hearted").remove();
                    var html = '<div id="unheart">';
                        html += '<a class="mb-0 px-0 py-1" href="#" id="liked">';
                        html += '<span class="ms-1">'+ response.count +'</span>';
                        html += '<input type="hidden" id="recipeId" value="{{ $recipe->id }}">';
                        html += '<img class="x1b0d499 xep6ejk" src="{{ asset("img/icons/icons8-white-heart-25.png") }}">';
                        html += '</a>';
                        html += '</div>';
                        html += '<div id="newHeartStatus"></div>';
                    $('#newHeartStatus').append(html);
                }
            }
        },

        fail: function(response) {
            Swal.fire({
                toast: true,
                position: 'bottom-end',
                showConfirmButton: false,
                timer: 5000,
                animation:true,
                showCloseButton: true,
                icon: 'error',
                title: response.message
            });
        }
        })
    })

My Controller;

         public function likes($id)
    {
        $user = User::find(auth()->user()->id);
        $recipe = Recipe::find($id);
        $reacter = $user->viaLoveReacter();

        if ($reacter->hasReactedTo($recipe, 'Like')) {
            $reacter->unreactTo($recipe, 'Like');
            $reactantFacade = $recipe->viaLoveReactant();
            $reactionLikeCounter = $reactantFacade->getReactionCounterOfType('Like');

            $response['success'] = true;
            $response['status'] = 'Disliked';
            $response['count'] = $reactionLikeCounter->count;
            $response['title'] = "You have Disliked this recipe!";
            $response['icon'] = "success";
            return response()->json($response);

        }else{
            $reacter->reactTo($recipe, 'Like');
            $reactantFacade = $recipe->viaLoveReactant();
            $reactionLikeCounter = $reactantFacade->getReactionCounterOfType('Like');

            $response['success'] = true;
            $response['status'] = 'Liked';
            $response['count'] = $reactionLikeCounter->count;
            $response['title'] = "You have Liked this recipe!";
            $response['icon'] = "success";
            return response()->json($response);

        }

    }


You can't find anything like this on the web. Figured this out for a week. Your package is great! 

I am able to use it also in my Laravel-wide search where it will show how many Likes my Recipe has.

**Reply by antonkomarev** [reply]
@iamB0rgy thank you for sharing! Is it a publicly available project?

**Reply by iamB0rgy** [reply]
@antonkomarev it is under development. Soon it will be in public.

**Reply by antonkomarev** [reply]
Feel free to share your project here

**Reply by iamB0rgy** [reply]
Sure. I will.
